### PR TITLE
ISSD-1364 Replace expando column

### PR DESCRIPTION
**Pull request by Andrefes565**

### CODE REVIEW

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 1]
```diff
@@ -2398,8 +2400,7 @@ private void _assertOrganizations2() throws Exception {
 		_assertAccountOrganizationRelsCount(organization, 0);
 	}
 
-	private void _assertPermissions() throws Exception {
-		_assertRoles();
+	private void _assertPermissions() throws Exception {_assertRoles();
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:2403]: why did you do that?

**Review by nNilton — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 2]
```diff
@@ -3166,6 +3170,19 @@ private void _addOrUpdateResourcePermissions(
...
+		for (ExpandoColumn expandoColumn :
+			expandoColumns) {
+
+			stringUtilReplaceValues.put(
+				"EXPANDO_COLUMN_ID:" +
+				expandoColumn.getName(),
+				String.valueOf(
+					expandoColumn.getColumnId()));
+		}
+
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:3185]: please move this behavior to a proper method responsible to handle expando columns

**Review by nNilton — CHANGES_REQUESTED** [review]

**Comment by nNilton** [comment]
please put the prefix on every single commit (ISSD-????)

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java [hunk 3]
```diff
@@ -3851,6 +3853,10 @@ private void _test2(SiteInitializer siteInitializer) throws Exception {
 	private LayoutPageTemplateEntryLocalService
 		_layoutPageTemplateEntryLocalService;
 
+	@Inject
+	private ExpandoColumnLocalService
+		_expandoColumnLocalService;
+
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java:3859]: please sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 4]
```diff
@@ -262,6 +264,7 @@ public BundleSiteInitializer(
 		LayoutCopyHelper layoutCopyHelper,
 		LayoutLocalService layoutLocalService,
 		LayoutPageTemplateEntryLocalService layoutPageTemplateEntryLocalService,
+		ExpandoColumnLocalService expandoColumnLocalService,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:267]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 5]
```diff
@@ -345,6 +348,7 @@ public BundleSiteInitializer(
 		_layoutLocalService = layoutLocalService;
 		_layoutPageTemplateEntryLocalService =
 			layoutPageTemplateEntryLocalService;
+		_expandoColumnLocalService = expandoColumnLocalService;
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:351]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 6]
```diff
@@ -1979,6 +1996,19 @@ private void _addOrUpdateExpandoColumns(ServiceContext serviceContext)
 				continue;
 			}
 
+			List<ExpandoColumn> expandoColumns =
+				_expandoColumnLocalService.getExpandoColumns(-1, -1);
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:2000]: you dont need to get all expandoColumns once again

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 7]
```diff
@@ -5054,6 +5084,7 @@ private void _updateLayoutSets(
 	private final LayoutLocalService _layoutLocalService;
 	private final LayoutPageTemplateEntryLocalService
 		_layoutPageTemplateEntryLocalService;
+	private final ExpandoColumnLocalService _expandoColumnLocalService;
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:5087]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtender.java [hunk 8]
```diff
@@ -143,7 +144,7 @@ public SiteInitializerExtension addingBundle(
 				_groupLocalService, _journalArticleLocalService, _jsonFactory,
 				_knowledgeBaseArticleResourceFactory,
 				_knowledgeBaseFolderResourceFactory, _layoutCopyHelper,
-				_layoutLocalService, _layoutPageTemplateEntryLocalService,
+				_layoutLocalService, _layoutPageTemplateEntryLocalService, _expandoColumnLocalService,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtender.java:147]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtender.java [hunk 9]
```diff
@@ -264,7 +265,7 @@ private void _addFile(File file) throws Exception {
...
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtender.java:268]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtension.java [hunk 10]
```diff
@@ -123,6 +124,7 @@ public SiteInitializerExtension(
 		LayoutCopyHelper layoutCopyHelper,
 		LayoutLocalService layoutLocalService,
 		LayoutPageTemplateEntryLocalService layoutPageTemplateEntryLocalService,
+		ExpandoColumnLocalService expandoColumnLocalService,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtension.java:127]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtension.java [hunk 11]
```diff
@@ -193,7 +195,7 @@ public SiteInitializerExtension(
 			journalArticleLocalService, jsonFactory,
 			knowledgeBaseArticleResourceFactory,
 			knowledgeBaseFolderResourceFactory, layoutCopyHelper,
-			layoutLocalService, layoutPageTemplateEntryLocalService,
+			layoutLocalService, layoutPageTemplateEntryLocalService, expandoColumnLocalService,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerExtension.java:198]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerFactoryImpl.java [hunk 12]
```diff
@@ -130,7 +131,7 @@ public SiteInitializer create(File file, String symbolicName)
 			_journalArticleLocalService, _jsonFactory,
 			_knowledgeBaseArticleResourceFactory,
 			_knowledgeBaseFolderResourceFactory, _layoutCopyHelper,
-			_layoutLocalService, _layoutPageTemplateEntryLocalService,
+			_layoutLocalService, _layoutPageTemplateEntryLocalService, _expandoColumnLocalService,
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerFactoryImpl.java:134]: sort

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerFactoryImpl.java [hunk 13]
```diff
@@ -263,6 +264,9 @@ protected void activate(BundleContext bundleContext) {
 		_layoutPageTemplateEntryLocalService;
 
 	@Reference
+	private ExpandoColumnLocalService
+		_expandoColumnLocalService;
+	@Reference
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/SiteInitializerFactoryImpl.java:269]: sort

**Review by nNilton — CHANGES_REQUESTED** [review]

**Comment by Andrefes565** [comment]
ci:reopen

**Comment by Andrefes565** [comment]
Reopen pull request

**Pull request closed without merging**